staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -231,12 +236,47 @@ func (s *SpdyRoundTripper) dial(req *http.Request) (net.Conn, error) {
 	return tlsConn, nil
 }
 
+func (s *SpdyRoundTripper) tlsDialWithDialFn(ctx context.Context, dialAddr string) (*tls.Conn, error) {
+	rawConn, err := s.DialFn(ctx, "tcp", dialAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	colonPos := strings.LastIndex(dialAddr, ":")
```

Comment:
What if dialAddr is a IPv6 address with no port?

Reply:
Good point. I'll switch to net.SplitHostPort.

This was copied from the tls stand lib (https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/crypto/tls/tls.go;l=138). Curious why it hasn't had any problem yet.

Reply:
is it possible to pass an address without port?

Reply:
it seem you can skip the address for tcp, but not the port https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/net/dial.go;l=278-297